I3DShapesTool.Lib/Model/I3DShape.cs

```diff
@@ -135,67 +135,84 @@ protected override void ReadContents(BinaryReader reader)
 
             if(Options.HasFlag(I3DShapeOptions.HasSkinningInfo))
             {
-                bool noBlendWeights = Options.HasFlag(I3DShapeOptions.NoBlendWeights);
-                bool blendWeights = !noBlendWeights;
+                bool singleBlendWeights = Options.HasFlag(I3DShapeOptions.SingleBlendWeights);
 
-                // Load blend weights first
-                if(blendWeights)
+                // based on how 3D engines usually do skeletal/skinned meshes, you always have 4 weights and 4 indices per vertex
+
+                BlendWeights = new float[VertexCount, 4];
+
+                int numIndices;
+
+                // Load blend weights first, if necessary
+                // usually all weights per vertex sum up to 1.0, which means, if you only have one index per vertex, the first weight would always have to be 1
+                if (singleBlendWeights)
                 {
-                    BlendWeights = new float[VertexCount, 4];
-                    for(int i = 0; i < BlendWeights.GetLength(0); i++)
+                    numIndices = 1;
+
+                    for(int i = 0; i < VertexCount; i++)
                     {
-                        for(int j = 0; j < BlendWeights.GetLength(1); j++)
+                        BlendWeights[i, 0] = 1.0f;
+                    }
+                }
+                else
+                {
+                    numIndices = 4;
+
+                    for(int i = 0; i < VertexCount; i++)
+                    {
+                        for(int j = 0; j < 4; j++)
                         {
                             BlendWeights[i, j] = reader.ReadSingle();
                         }
                     }
                 }
 
                 // Load blend indices
-                // If weights exist, we will have 4 index slots per vertex, otherwise just 1.
-                BlendIndices = new byte[VertexCount, blendWeights ? 4 : 1];
-                for(int i = 0; i < BlendIndices.GetLength(0); i++)
+                BlendIndices = new byte[VertexCount, 4];
```

Comment:
Should be numIndices here and not just "4"

Reply:
Well... There are two cases in general and 3 cases in Farming Simulator.
1. no blending/skinning at all, because it is a static mesh
2. it is a skeletal/skinned mesh, so there are always 4 blend indices and weights (even if you don't need all, the graphics math is the same for if you have 1, 2, 3 or 4)
3. FS uses the special case of 1 blend index to merge multiple meshes into one. This is a major grievance for modders, as they have to manually separate those, because OBJ export only contains the merged mesh
If you want to export to a graphics format like FBX or gltf, you'll have to deal with 4 indices anyway.